### ISSUE TITLE
OSX and Linux builds are timing out before completion

### ISSUE THREAD

**Issue by jcagme** [dotnet/buildtools#1899]
@jcagme commented on [Tue Feb 06 2018](https://github.com/dotnet/corefx/issues/26897)

@dotnet-mc-bot commented on [Tue Feb 06 2018](https://github.com/dotnet/core-eng/issues/2590)

Opened on behalf of @jcagme

Warnings: 2
Status Message: failed

Build : Master - 20180206.04 (Product Build)
Failing configurations:
- OSX
  - x64
[Mission Control Build Info](https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/build~2Fproduct~2F/build/20180206.04/workItem/Orchestration/analysis/external/Link)

The job running on agent DDVSOMACAGE006 has exceeded the maximum execution time of 60.

**Comment by jcagme** [comment]
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?buildId=1355093
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/index?buildId=1356276
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/index?buildId=1354670
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/index?buildId=1354578
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?buildId=1351230
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?&buildId=1351234

**Comment by danmosemsft** [comment]
Seems the build completes but it doesn't schedule the test jobs - at least ont hte couple I looked at:
```
2018-02-05T14:02:30.2228750Z ##[error]The operation was canceled.
2018-02-05T14:02:30.2375840Z ##[section]Finishing: Create Helix Test Jobs
```

**Comment by joperezr** [comment]
Yeah the build is timing out since the full build should complete in 60 mins and it is not anymore. This is happening because of the recent changes we made on non-Windows to build with the CLI's msbuild. @weshaggard and I believe this might be because crossgen is not running on the tools anymore, so I'm currently trying to fix that to see if that will make our builds fast again.

**Comment by kasper3** [comment]
@joperezr, shouldn't CLI pack crossgen'd roslyn in first place? i.e. https://github.com/dotnet/cli/pull/8542

**Comment by joperezr** [comment]
Yes, but we are using a newer version of Roslyn than the one that comes with the CLI since we need latest Roslyn features in order to build our product. I have done a bit more investigation and it doesn't seem to be a crossgen problem actually, but a compiler server problem instead. I will update this thread once we know more details.

**Comment by joperezr** [comment]
This one was fixed by #28272
